sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1845,6 +1845,17 @@ object GpuOverrides extends Logging {
           // passing the already converted strf string for a little optimization
           GpuFromUnixTime(lhs, rhs, strfFormat)
       }),
+    expr[FromUTCTimestamp](
+      "Render the input UTC timestamp in the input timezone",
+      ExprChecks.binaryProject(TypeSig.TIMESTAMP, TypeSig.TIMESTAMP,
+        ("timestamp", (TypeSig.STRING + TypeSig.TIMESTAMP)
+          .withPsNote(TypeEnum.STRING, "A limited number of timestamp formats are supported"),
+          TypeSig.STRING + TypeSig.TIMESTAMP),
```

Comment:
Is STRING really supposed to be here?  The code does not handle an input of string, and the Spark code states the timestamp parameter must be TimestampType.

Reply:
Spark code `FromUTCTimestamp` inherits from `UTCTimestamp`, which supports string:
```
sealed trait UTCTimestamp extends BinaryExpression with ImplicitCastInputTypes with NullIntolerant {
  override def inputTypes: Seq[AbstractDataType] = Seq(TimestampType, StringType)
  override def dataType: DataType = TimestampType
```

Reply:
Actually we don't need to do that. And we don't need to support string input because it will be implicitly casted into timestamp:
```
val df = Seq(("1999-12-24 03:58:55.123456Z", "UTC"), ("2020-12-12 01:02:33.123456Z", "UTC")).toDF("v", "z")
df: org.apache.spark.sql.DataFrame = [v: string, z: string]

scala> df.repartition(1).selectExpr("from_utc_timestamp(v, 'UTC')").show(false)
....
 *Expression <Alias> cast(from_utc_timestamp(cast(v#68 as timestamp), UTC) as string) AS from_utc_timestamp(CAST(v AS TIMESTAMP), UTC)#75 will run on GPU
...
```

Reply:
> Spark code FromUTCTimestamp inherits from UTCTimestamp, which supports string

It only supports TimestampType.  That sequence is not a list of possible types for each parameter, it is the 1-to-1 mapping of types corresponding to each input parameter.  The first parameter (timestamp) must be TimestampType and the second parameter (timezone) must be a String.

Reply:
Ah I see. Indeed I thought the list is all supported types.